File: CS50x/Week6/pset6/sentimental-mario-more/mario.py
# Author: Eskabore
# Purpose: Printing a pyramid with a gap in between
# Language: Python

from cs50 import get_int


def main():

    # Prompt user for height of the Pyramid
    height = get_int("Enter a number between 1 to 8: ")
    while height < 1 or height > 8:
        height = get_int("Type an integer between 1 to 8: ")

    # Outer loop used to point to current row
    for i in range(1, height+1):
        # Inner loop printing left-aligned spaces
        for j in range(height, i, -1):
            print(" ", end="")
        # Inner loop printing left-aligned pyramid
        for k in range(1, i+1):
            print("#", end="")
        # Print gap
        print("  ", end="")
        # Inner loop printing right-aligned pyramid
        for i in range(1, i+1):
            print("#", end="")
        # Insert a new line
        print()


if __name__ == "__main__":
    main()

File: CS50x/Week1/pset1/mario/mario-more/mario.c
// Author: Eskabore
// Purpose: Printing a pyramid with a gap in between
// Language C
#include <cs50.h>
#include <stdio.h>

int main(void)


{

    int height;

    // Prompt user for height of the pyramid
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Outer loop used to point to current row
    for (int i = 1; i <= height; i++)
    {
        // Inner loop printing left-aligned spaces
        for (int j = height; j > i; j--)
        {
            printf(" ");
        }

        // Inner loop printing left-aligned pyramid
        for (int k = 1; k <= i; k++)
        {
            printf("#");
        }

        // Print gap
        printf("  ");

        // Inner loop printing right-aligned pyramid
        for (int l = 1; l <= i; l++)
        {
            printf("#");
        }

        // Insert a new line
        printf("\n");
    }
}